flowmachine/flowmachine/core/cache.py

```
@@ -0,0 +1,384 @@
+            f"Table '{table_schema}.{table_name}' does not exist on this connection."
+        )
+
+
+def size_of_cache(connection: "Connection") -> int:
+    """
+    Get the total size in bytes of all cache tables.
+
+    Parameters
+    ----------
+    connection : "Connection"
+
+    Returns
+    -------
+    int
+        Number of bytes in total used by cache tables
+
+    """
+    sql = """SELECT sum(table_size(tablename, schema)) as total_bytes 
+        FROM cache.cached  
+        WHERE NOT cached.class='Table'"""
+    cache_bytes = connection.fetch(sql)[0][0]
+    return 0 if cache_bytes is None else int(cache_bytes)
+
+
+def get_max_size_of_cache(connection: "Connection") -> int:
+    """
+    Get the upper limit set in FlowDB for the cache size, in bytes.
+
+    Parameters
+    ----------
+    connection : "Connection"
+
+    Returns
+    -------
+    int
+        Number of bytes in total available cache tables
+
+    """
+    sql = "SELECT cache_max_size()"
+    return int(connection.fetch(sql)[0][0])
+
+
+def get_cache_half_life(connection: "Connection") -> float:
+    """
+    Get the current setting for cache half-life.
+
+    Parameters
+    ----------
+    connection : "Connection"
+
+    Returns
+    -------
+    float
+        Cache half-life setting
+
+    """
+    sql = "SELECT cache_half_life()"
+    return float(connection.fetch(sql)[0][0])
+
+
+def set_cache_half_life(connection: "Connection", cache_half_life: float) -> None:
+    """
+    Set the cache's half-life.
+
+    Parameters
+    ----------
+    connection : "Connection"
+    cache_half_life : float
+        Setting for half-life
+
+    Notes
+    -----
+
+    Changing this setting without flushing the cache first may have unpredictable consequences
+    and should be avoided.
+    """
+    sql = (
+        f"UPDATE cache.cache_config SET value='{cache_half_life}' WHERE key='half_life'"
+    )
+    with connection.engine.begin() as trans:
+        trans.execute(sql)
```

What happens (here or in other functions) if the user passes a wrong parameter (e.g. something that can't be cast to a float)? Do we need to raise a meaningful error or are we ok for the `sqlalchemy`/`psycopg2` error to bubble up?